src/praisonai-train/praisonai_train/data/generate.py

```
@@ -170,3 +238,13 @@ def generate_dataset(
                 yield row
             if progress_callback:
                 progress_callback(done, total, kept)
+            # In-loop continuation guard (billing/sponsorship self-check): every
+            # ``check_every`` completions, if the predicate says stop, halt cleanly
+            # via the same shutdown path as the stop-file breaker.
+            if guard is not None and done % check_every == 0 and not guard():
+                try:
+                    open(stop_file, "w").close()
+                except Exception:
+                    pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)